GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -239,15 +239,18 @@ public String getName() {
     }
 
     /**
-     * Set the (optional) name of the object.
+     * Set the (optional) name of the object. Scene object names are not needed:
+     * they are only for the application's convenience.
      * 
-     * Scene object names are not needed: they are only for the application's
-     * convenience.
-     * 
-     * @param name
-     *            Name of the object.
+     * @param name Name of the object.
+     * @throws IllegalArgumentException if input parameter is null or empty.
      */
     public void setName(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
Imho, it won't be cheap if we do trim() check on every input string since it doesn't only check for space but also check for some other characters, and even finally generate a new String. We may discuss if it is necessary to do such check and if necessary we can change Preconditions.checkStringNotNullOrEmpty().  We should unify the standards for each similar check and finally make code consistent and reduce duplicate codes.